src/stripeWorkspaceState.ts

```diff
@@ -89,9 +91,48 @@ export function addLogDetails(
   extensionContext.workspaceState.update(logDetailsKey, logDetailsMap);
 }
 
-export function retrieveLogDetails(extensionContext: vscode.ExtensionContext, logId: string) {
+export async function retrieveLogDetails(extensionContext: vscode.ExtensionContext, logId: string, daemonClient: StripeCLIClient) {
   const logDetailsMap = getLogDetailsMap(extensionContext);
-  return logDetailsMap.get(logId);
+  const logDetails = logDetailsMap.get(logId);
+
+  // if insight has not been retrieved or previously failed to be retrieved, then retrieve it
+  if (!('insight' in logDetails) || logDetails.insight.includes('Failed to retrieve insight') || logDetails.insight.includes('Please check back later')) {
+    const insight = await getIntegrationInsight(logId, daemonClient);
+    logDetails.insight = insight;
```

Comment:
Nit: I wonder if we should avoid setting this field if there's no insight. So the user won't see an empty string. What do you think?

Reply:
currently there could be empty insight. i am trying to distinguish between `insight not retrieved yet` and `actual empty insight`, so we dont keep retrieving insight for the same log.